src/main/java/com/configcat/ConfigCatLogger.java

```diff
@@ -2,54 +2,62 @@
 
 import org.slf4j.Logger;
 
+import java.util.function.Function;
+
 class ConfigCatLogger {
     private final Logger logger;
     private final LogLevel logLevel;
     private final ConfigCatHooks configCatHooks;
+    private final Function<FilterFunctionParameters, Boolean> filterOutFunction;
 
-    public ConfigCatLogger(Logger logger, LogLevel logLevel, ConfigCatHooks configCatHooks) {
+    public ConfigCatLogger(Logger logger, LogLevel logLevel, ConfigCatHooks configCatHooks, Function<FilterFunctionParameters, Boolean> filterOutFunction) {
         this.logger = logger;
         this.logLevel = logLevel;
         this.configCatHooks = configCatHooks;
+        this.filterOutFunction = filterOutFunction;
     }
 
     public ConfigCatLogger(Logger logger, LogLevel logLevel) {
-        this(logger, logLevel, null);
+        this(logger, logLevel, null, null);
     }
 
     public ConfigCatLogger(Logger logger) {
         this(logger, LogLevel.WARNING);
     }
 
     public void warn(int eventId, String message) {
-        if (this.logLevel.ordinal() <= LogLevel.WARNING.ordinal()) {
+        if (this.logLevel.ordinal() <= LogLevel.WARNING.ordinal() && !checkFilterOut(new FilterFunctionParameters(eventId, message, LogLevel.WARNING))) {
             this.logger.warn("[{}] {}", eventId, message);
         }
     }
 
     public void error(int eventId, String message, Exception exception) {
         if (this.configCatHooks != null) this.configCatHooks.invokeOnError(message);
-        if (this.logLevel.ordinal() <= LogLevel.ERROR.ordinal()) {
+        if (this.logLevel.ordinal() <= LogLevel.ERROR.ordinal() && !checkFilterOut(new FilterFunctionParameters(eventId, message, LogLevel.ERROR, exception))) {
```

Comment:
We could reduce code repetition if we moved the log level check into the shared check filter method:

```suggestion
        if (!filter(eventId, message, LogLevel.ERROR, exception)) {
```

Then the `filter` method would look like this:

```java
    private boolean filter(int eventId, LogLevel logLevel, String message, Exception exception) {
        return this.logLevel.ordinal() <= logLevel.ordinal && (this.filterOutFunction == null || this.filterOutFunction.apply(/* ... */));
    }
```

Another thing to mention here that in my experience usually a true return value means to keep the item, a false return value to omit it.